### PR TITLE
samples: modules: thrift: improve Thrift documentation

### PR DESCRIPTION
**Pull request by cfriedt**
* add thrift-layers image
 * ensure Thrift documents are globbed in `modules/index.rst`
 * add a "What is Thrift?" section to the Thrift sample

### CODE REVIEW

File: samples/modules/thrift/hello/doc/index.rst [hunk 1]
```diff
@@ -1,14 +1,47 @@
...
+
+.. _Apache Software Foundation:
+    https://www.apache.org
 
 Overview
 ********
 
-This sample application includes a client a server implementing the RPC
+This sample application includes a client and server implementing the RPC
 interface described in ``thrift/hello.thrift``. The purpose of this
```
Comment (mbolivar-nordic) [inline samples/modules/thrift/hello/doc/index.rst:43]: Non-blocking:

```
:zephyr_file:`samples/modules/thrift/hello/hello.thrift`
```

File: samples/modules/index.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ External Module Samples
 #######################
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
```
Comment (mbolivar-nordic) [inline samples/modules/index.rst:7]: Did you verify that this doesn't produce too much noise in the [samples index](https://docs.zephyrproject.org/latest/samples/index.html)?

**Review by mbolivar-nordic — COMMENTED** [review]
Thanks for the quick fix!

**Comment by mbolivar-nordic** [comment]
@nashif @stephanosio something is wrong with the labeling algorithm here:

> zephyrbot added the manifest-thrift label 15 hours ago

This isn't touching the manifest

File: samples/modules/index.rst (hunk 2 above)
Comment (cfriedt) [reply samples/modules/index.rst:7]: I did not - I could move it back, but I think I would need to move stuff out of the `doc/` subdirectory to compensate, which is fine - I just need to then go to a few social media locations and update comments 😅

File: samples/modules/thrift/hello/doc/index.rst (hunk 1 above)
Comment (cfriedt) [reply samples/modules/thrift/hello/doc/index.rst:43]: Will fix

File: samples/modules/index.rst (hunk 2 above)
Comment (mbolivar-nordic) [reply samples/modules/index.rst:7]: @kartben do you have an opinion here?

Comment (kartben) [reply samples/modules/index.rst:7]: I think I'd rather get rid of the doc/ folder, frankly, as most samples tend to not have one and it's nice for the "See it on GitHub" link thingy to take one as close as possible to the actual source of a particular sample IMO.

That being said I can live with an update to the TOC depth, especially if you have existing links you don't want to break (in addition to pages from 3.3 which will 404 when people click on the "see the most recent version of this page banner).

Comment (mbolivar-nordic) [reply samples/modules/index.rst:7]: > (in addition to pages from 3.3 which will 404 when people click on the "see the most recent version of this page banner).

Are you sure? One problem this PR addresses is that this page is not present in the HTML for the toctree rooted here.

And anyway, that can be fixed with doc/_scripts/redirects.py if so.

Comment (kartben) [reply samples/modules/index.rst:7]: I thought Chris was saying he had social posts that may be directly linking to https://docs.zephyrproject.org/3.3.0/samples/modules/thrift/hello/doc/index.html -- page isn't mounted in the toc, but it's still there. So if we move it, yes, redirects will need to be updated (thanks for reminding me this exists :-))

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by cfriedt** [comment]
@kartben - does this look better? Are you able to approve?

**Review by stephanosio — APPROVED** [review]

**Pull request merged**